jax/_src/lax/lax.py

```diff
@@ -1111,7 +1115,7 @@ def stop(x):
       return x  # only bind primitive on inexact dtypes, to avoid some staging
   return tree_map(stop, x)
 
-def reduce_precision(operand, exponent_bits, mantissa_bits):
+def reduce_precision(operand: T, exponent_bits: int, mantissa_bits: int) -> T:
```

Comment:
Same here.

Reply:
I added an overload for `float`.  This function doesn't accept integers or complex dtypes.